pkg/prometheus/operator.go

```diff
@@ -1929,6 +1922,9 @@ func (c *Operator) selectPodMonitors(p *monitoringv1.Prometheus) (map[string]*mo
 	// Selectors might overlap. Deduplicate them along the keyFunc.
 	res := make(map[string]*monitoringv1.PodMonitor)
 
+	if p.Spec.PodMonitorSelector == nil {
```

Comment:
It would be nice to instead change the types so this becomes a literal instead of a pointer but that might break consumers. Do we care?

Reply:
Could you verify what would happen with an object currently has nil and we change the type to use the literal? If it defaults to the empty literal then I'm happy with changing this. As in if forward compatibility is guaranteed I'm totally ok with this. If we end up doing this, can we make sure to do the same with `probeSelector`?

Reply:
You mean if the json serialization doesn't have the field? It will default to the empty literal because it can not be nil anymore. Once we update the Prometheus, we will insert a `fieldName: {}` into it.

I was mostly thinking about possible downstream consumers of the api if there are any.